packages/datastore/src/request.js

```diff
@@ -1057,7 +1083,34 @@ DatastoreRequest.prototype.request_ = function(protoOpts, reqOpts, callback) {
     };
   }
 
-  this.request(protoOpts, reqOpts, callback);
+  datastore.auth.getProjectId(function(err, projectId) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var gaxClient = datastore.api[config.client];
+
+    var clientOptions = extend({
+      servicePath: datastore.baseUrl_,
+      port: datastore.port_
+    }, datastore.options);
+
+    if (datastore.customEndpoint_) {
+      clientOptions.sslCreds = grpc.credentials.createInsecure();
```

Comment:
I'm in the middle of some long running tests, so I can't flip to this branch, but I believe this is because of this code: https://github.com/googleapis/gax-nodejs/pull/129/files#diff-a5e6d94eb9e5535646d0ffb59b877292L82